charts/kyverno/values.yaml

```diff
@@ -216,7 +216,7 @@ config:
   - '[TokenReview,*,*]'
   - '[SubjectAccessReview,*,*]'
   - '[SelfSubjectAccessReview,*,*]'
-  - '[*,kyverno,kyverno*]'
+  - '[*,{{ include "kyverno.namespace" . }},kyverno*]'
```

Comment:
Thanks for the fix @eddycharly!
@realshuting - yes, it's related to #3339, #3304 and #2356. There's one last thing that can be improved here from my understanding.

The value of `kyverno*` in the third field _(which represents the resource name)_ should be handled for release names other than `kyverno` as well, which can probably be dealt with by using the `fullnameOverride` variable in the helm chart.
@eddycharly can you please look into this and verify if it fixes the issue you mentioned here:
https://kubernetes.slack.com/archives/CLGR9BJU9/p1646756104825019?thread_ts=1645650633.460769&cid=CLGR9BJU9

Reply:
It's not as simple as using `fullnameOverride`, it will probably require a bit of templating. I will look into it asap.

Reply:
BTW any reason why we want to exclude resources in the kyverno ns ?

Reply:
At first sight, it's going to be difficult. There are a couple of patterns used throughout all the resource templates `kyverno.fullname`, `kyverno.serviceAccountName`, `kyverno.serviceName` ...
And some can even be user defined (`kyverno.serviceAccountName` for example).

I will check but it's not that trivial.

Reply:
We would not want the resources created by Kyverno to be affected by any of the cluster / namespaced policies. In case of policies having `validationFailureAction: enforce`, there might be a few scenarios where the Kyverno resources would be prohibited from being modified or spinning up. The `kyverno*` wildcard match ensures that in the default setting, none of the Kyverno resources would be affected by any policy.

Reply:
> At first sight, it's going to be difficult. There are a couple of patterns used throughout all the resource templates `kyverno.fullname`, `kyverno.serviceAccountName`, `kyverno.serviceName` ... And some can even be user defined (`kyverno.serviceAccountName` for example).
> 
> I will check but it's not that trivial.

From our end we can surely mention these details in the documentation itself, while also mentioning that in case the user makes any changes to such variables, they should understand that it's their responsibility to modify the `resourceFilters` accordingly.

Reply:
Unfortunately when installed with helm, a resource name can start with something else than `kyverno`, `kyverno*` will not match.

> We would not want the resources created by Kyverno to be affected by any of the cluster / namespaced policies.

What are the advantages/pitfalls of allowing policies to apply to kyverno resources ?

Reply:
The only advantage I can think of right now is that we can probably set our own standards for deploying / implementing Kyverno within the cluster where any unintended changes could be either prohibited or patched with appropriate values.
The major disadvantage would be that in a scenario the user modifies or restarts any existing Kyverno resource while violating any of the existing enforced policies with matching kind, it might result in unexpected behaviours or even failures.

Reply:
The best I could come up with https://github.com/kyverno/kyverno/pull/3361 :man_shrugging: